R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,677 @@
+   "metadata": {},
+   "source": [
+    "## Image segmentation\n",
+    "\n",
+    "### Visual\n",
+    "\n",
+    "We are using four steps for the segmentation:\n",
+    "- Thresholing using the [thresh](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2/topics/thresh) function to get a binary image.\n",
+    "- Filtering to remove noise, using [medianFilter](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2/topics/medianFilter).\n",
+    "- [fillHull](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2/topics/fillHull) to close holes in the segmented objects.\n",
+    "\n",
+    "Note: Up to now we still have a binary image (the pixel array only consists of 0's (background) and 1's (objects), therefore we need:\n",
+    "\n",
+    "- [bwlabel](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2/topics/bwlabel) to distinguish the single objects from each other. It assigns a different number to each object (first object will get pixel values '1', the second pixel values '2', etc.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Threshold\n",
+    "threshImg <- thresh(ebimage, w=10, h=10, offset=0.01)\n",
+    "\n",
+    "# Remove noise\n",
+    "threshImg <- medianFilter(threshImg, size=3)\n",
+    "\n",
+    "# Fill holes\n",
+    "threshImg <- fillHull(threshImg)          \n",
+    "\n",
+    "# Distinguish objects\n",
+    "threshImg <- bwlabel(threshImg)\n",
+    "\n",
+    "# Show the segmented image\n",
+    "EBImage::display(colorLabels(threshImg))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Modify the parameters of the thresh function w, h and offset to get a better segmentation. Goal: The two centrioles should be detected as two separate objects. We don't have to get rid off all the noise, but the two centrioles should be the largest objects detected. (w=15, h=15, offset=0.1 seems to work pretty well)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compute the features (measurements)\n",
+    "\n",
+    "Use the [computeFeatures](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2/topics/computeFeatures) methods to calculate some properties of the objects."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "shapes = computeFeatures.shape(threshImg)\n",
+    "moments = computeFeatures.moment(threshImg)\n",
+    "\n",
+    "# merge the two dataframes together into one 'features' dataframe\n",
+    "features <- merge(shapes, moments, by=0, all=TRUE)\n",
+    "features"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the further analysis we need the object sizes: 's.area', 's.perimeter', especially 's.radius.mean' (~ diameter).\n",
```

Comment:
``For the further analysis...`` -> ``For further analysis...``

``...object sizes`` -> ``...object metrics describing the size:``


``, especially...`` -> ``...and especially...``